### ISSUE TITLE
bug about tags function of ansible python api 

### ISSUE THREAD

**Issue by bruceliu01** [ansible/ansible#38315]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

tags of ansible python api

##### ANSIBLE VERSION

```
[root@controller01 ~]# ansible --version
ansible 2.3.0.0
config file = /etc/ansible/ansible.cfg
configured module search path = Default w/o overrides
python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

##### OS / ENVIRONMENT

centos python2.7

##### SUMMARY

It can not work right when I use tags feature of ansible python api. Ansible command can work right.
##### STEPS TO REPRODUCE

python playbook.py
the following is fragment of playbook.py
```
if __name__ =='__main__':
    play_book=AnsiblePlaybook(["/home/liux/ntp_server_ip.yml"], **tags=["get"]**, extra_vars=run_data)
    play_book.run()
    play_book.get_result()
```

```yaml
...
playbook for test
---
- name: NTP server metadata
  hosts: computers
  gather_facts: False
  tags:
    - MetaData
  vars: 
    name_cn: "NTP服务器"
    name_en: "NTP Server"
    role: CONTROLLER_HA
    params:
      - param1:
        name_cn: "NTP服务器IP" 
        name_en: "NTP server IP"
        type: string
        range: null
        default: null

- name: play with tag "set"
  hosts: computers
  gather_facts: False
  tags:
    - set
  tasks:
    - name: copy chrony.conf file 
      template:
        src: /home/liux/test
        dest: /home/test
        mode: 0644
        force: yes
    - name: append ntp server
      lineinfile:
        dest: /home/test
        insertafter: "# insert server"
        line: "server {{ item }}"
      with_items: "{{ NTP_SERVERS }}"

- name: play with tag "get"
  hosts: computers
  gather_facts: False
  tags:
    - get
  tasks:
    - shell: "grep 'server ' /home/test | awk '{print $2}'"
```

##### EXPECTED RESULTS

I wish it just run the play with tag "set"
```
===============================================================================
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++run the playplay with tag "set":
2018-04-05 12:35:45.905530 (delta: 0.063578)         elapsed: 0.067417 ********
===============================================================================
```
##### ACTUAL RESULTS

It runs all the play in the playbook
```
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++run the playNTP server metadata:
2018-04-05 12:35:45.841896 (delta: 0.003783)         elapsed: 0.003783 ********
===============================================================================
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++run the playplay with tag "set":
2018-04-05 12:35:45.905530 (delta: 0.063578)         elapsed: 0.067417 ********
===============================================================================
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++run the playplay with tag "get":
2018-04-05 12:35:45.910583 (delta: 0.005002)         elapsed: 0.07247 *********
===============================================================================
2018-04-05 12:35:45.913458 (delta: 0.002836)         elapsed: 0.075345 ********
2018-04-05 12:35:46.429407 (delta: 0.515907)         elapsed: 0.591294 ********
===============================================================================
```

```

```

**Comment by ansibot** [comment]
Files identified in the description:
None

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by bruceliu01** [comment]
ansbile run all the plays that Playbook.get_plays() return. But this method return all the play in the play book:
```
\ansible\playbook\__init__.py
def get_plays(self):
        return self._entries[:]
```

**Comment by bcoca** [comment]
List Information
================

Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

This appears to be a developer question, and we'd like to direct these kinds of things to either the mailing list or the IRC channel.

   * IRC: #ansible-devel on irc.freenode.net   
   * IRC: #ansible on irc.freenode.net   
   * mailing list: https://groups.google.com/forum/#!forum/ansible-devel
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

If you can stop by there, we'd appreciate it.  This allows us to keep the issue tracker for bugs, pull requests, RFEs and the like.

Thank you once again and we look forward to seeing you on the list or IRC.  Thanks!

**Comment by bruceliu01** [comment]
I have tried the IRC and mail list you gived me. However I can not visit these sites in China.
Do I have other choice?
I would describe the bug ,that I proposed on github, to you.
I have two plays in playbook, one with tag A, and another with tag B. I run the play with ansible python api, and I use the tag function in the options, that is a param for the class PlaybookExecutor.
If I run one play one time, it works right. But if I run play with tag A, and run play with tag B at once, it can not work right.
expect: run play with tag A; run play with tag B


actually: run play with tag A; run play with tag A and run play with tag B
It seems that it will run all the plays that are executed before.




发自网易邮箱大师


On 04/5/2018 22:11，Brian Coca<notifications@github.com> wrote：

Closed #38315.
